src/rime/gear/charset_filter.cc

```diff
@@ -79,10 +130,22 @@ bool CharsetFilterTranslation::LocateNextCandidate() {
 
 // CharsetFilter
 
-bool CharsetFilter::FilterText(const string& text, const string& charset) {
-  if (charset.empty()) return !contains_extended_cjk(text);
+bool CharsetFilter::FilterText(const string& text, const string& charset_with_parameter) {
+  vector<string> charset_arguments_vector;
+  boost::split(charset_arguments_vector, charset_with_parameter, boost::is_any_of("+"));
+  if (charset_arguments_vector.size() == 0) {
```

Comment:
This won't work. I guess `split` would return non-empty vector `[""]` if string is empty.